test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs

```diff
@@ -1513,6 +1942,100 @@ public class Class1
+        [Fact]
+        public void PackCommand_WarningDependencyVersionNotSpecifiedInJson()
+        {
+            var msbuild = Path.Combine(
+                    Environment.GetFolderPath(Environment.SpecialFolder.Windows),
+                    @"Microsoft.NET\Framework\v4.0.30319\msbuild.exe");
+            if (!File.Exists(msbuild))
+            {
+                return;
+            }
+
+            var nugetexe = Util.GetNuGetExePath();
+
+            using (var workingDirectory = TestFileSystemUtility.CreateRandomTestFolder())
+            {
+                var proj1Directory = Path.Combine(workingDirectory, "proj1");
+
+                // create project 1
+                Util.CreateFile(
+                    proj1Directory,
+                    "proj1.csproj",
+@"<Project ToolsVersion='4.0' DefaultTargets='Build'
+    xmlns='http://schemas.microsoft.com/developer/msbuild/2003'>
+  <PropertyGroup>
+    <OutputType>Library</OutputType>
+    <Config Condition=" + "\" '$(Config)' == ''\"" + @">Debug</Config>
+    <TargetFrameworkVersion>v4.0</TargetFrameworkVersion>
+  </PropertyGroup>
+  <PropertyGroup Condition=" + "\"'$(Config)'=='Debug'\"" + @">
+    <OutputPath>debug_out</OutputPath>
+  </PropertyGroup>
+  <PropertyGroup Condition=" + "\"'$(Config)'=='Release'\"" + @">
+    <OutputPath>release_out</OutputPath>
+  </PropertyGroup>
+  <ItemGroup>
+    <Compile Include='proj1_file1.cs' />
+  </ItemGroup>
+  <ItemGroup>
+    <Content Include='proj1_file2.txt' />
+  </ItemGroup>
+  <Import Project='$(MSBuildToolsPath)\Microsoft.CSharp.targets' />
+</Project>");
+                Util.CreateFile(
+                    proj1Directory,
+                    "proj1_file1.cs",
+@"using System;
+
+namespace Proj1
+{
+    public class Class1
+    {
+        public int A { get; set; }
+    }
+}");
+                Util.CreateFile(
+                    proj1Directory,
+                    "proj1_file2.txt",
+                    "file2");
+
+                Util.CreateFile(
+                    proj1Directory,
+                    "project.json",
+                    @"{
+  ""version"": ""1.0.0"",
+  ""authors"": [ ""feiling"" ],
+  ""owners"": [ ""feiling"" ],
+  ""requireLicenseAcceptance"": ""false"",
+  ""description"": ""description"",
+  ""copyright"": ""Copyright 2013"",
+  ""dependencies"": {
+    ""json"": { }
+  }
+}");
+                var r = CommandRunner.Run(
+                    msbuild,
+                    proj1Directory,
+                    "proj1.csproj /p:Config=Release",
+                    waitForExit: true);
+
+                // Act
+                r = CommandRunner.Run(
+                    nugetexe,
+                    proj1Directory,
+                    "pack project.json",
+                    waitForExit: true);
+                Assert.Equal(1, r.Item1);
+
+                // Assert
+                Assert.Contains("Package dependencies must specify a version range.", r.Item3);
+            }
+        }
```

Comment:
Maybe there's a reasonable way to share most of the code between the project.json and the .csproj version of these tests.